src/setuptools_scm/config.py

```diff
@@ -93,6 +95,11 @@ def __init__(
         self.git_describe_command = git_describe_command
         self.dist_name = dist_name
 
+        if search_parent_directories:
```

Comment:
the search should happen in the wd parse functions, else the config object is misconfigured, and that makes it harder to reason about

also this is a git only helper, so currently it is broken for non- git like mercurial

Reply:
lets put it on config as a value, then the parse function can deal with the details per scm

Reply:
Honestly I'm a little bit lost in finding the parse functions ;)

Reply:
Ah maybe I get it. Do you mean something like `dump_version`?

Reply:
i meant the parse function in the git module (right around the place that does the call to from potential workdir

Reply:
I added it to `git.parse` and that works well. What I however don't understand is that I don't think that this function is called from `get_version`?